lib/web/apiserver.go

```diff
@@ -1061,13 +1061,20 @@ func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprou
 		return client.LoginFailedRedirectURL
 	}
 
+	// provide a suggested redirect URL in the request
+	var proxyAddr string
+	if len(h.cfg.ProxyPublicAddrs) > 0 {
```

Comment:
If there are multiple public addresses, can we ensure we are not redirecting the user to a different proxy than the one they are reaching out to?

Reply:
`h.cfg.ProxyPublicAddrs` is pointing to a specific proxy's list of public addrs, so all of them should point to the same place.

Also, if I understand correctly, we always use the first `ProxyPublicAddr`, and the other addresses are added as principals in certificates generated by that proxy.

For example, if you set `public_addr: [proxy.example.com, localhost]` and then try `tsh -d login --proxy=localhost:3080`, you'll see in the logs that you're actually connecting to `proxy.example.com:3080`.

Reply:
I think it is possible to have both say `proxy.company.com` and `teleport.company.com`.

Reply:
Hmm you're right, good catch. In the webui it will now redirect you to the original address if a proxy has multiple public addresses, but `tsh login` still always uses the first proxy address for a given proxy.